packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1465,22 +1528,27 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
       },
     );
 
+    if (widget._inputDecoration != null) {
+      result = InputDecorator(
+        decoration: widget._inputDecoration!,
+        isEmpty: widget._isEmpty,
+        isFocused: widget._isFocused,
+        child: result,
+      );
+    }
+
     return Semantics(
       button: true,
       child: Actions(
         actions: _actionMap,
-        child: Focus(
+        child: InkWell(
```

Comment:
I looked into how other widgets use `InkWell` and noticed that `ButtonStyleButton` uses `ConstrainedBox` and `Material` along with `InkWell`. I don't fully understand how `MaterialStateProperty` works, but I think this is how `InkWell` is supposed to be used.

https://github.com/flutter/flutter/blob/4e468f0122a9b814720d2a09176dd1bbaeeb8bc8/packages/flutter/lib/src/material/button_style_button.dart#L338-L349

Reply:
`ButtonStyleButton` buttons have fixed initial size that's the reason `ButtonStyleButton` uses `ConstrainedBox`

Such as `TextButton`
https://github.com/flutter/flutter/blob/8017883a5dfd13d709c4bd04d617f54a6cb2271f/packages/flutter/lib/src/material/text_button.dart#L287

Material is not required to be wrapped for InkWell, as long as you have a Material ancestor such as Scaffold, it is enough for InkWell and Material modifies those properties based on button requirements too
https://github.com/flutter/flutter/blob/8017883a5dfd13d709c4bd04d617f54a6cb2271f/packages/flutter/lib/src/material/list_tile.dart#L1214


I am not sure either is required here

Reply:
If that's the case, it needs to be mentioned that `DropdownButton` needs Material as its ancestor.

https://github.com/flutter/flutter/blob/8017883a5dfd13d709c4bd04d617f54a6cb2271f/packages/flutter/lib/src/material/list_tile.dart#L552-L553

Reply:
Good catch, I will add that